src/parquet/column_reader.cc

```diff
@@ -212,6 +121,9 @@ bool ColumnReader::ReadNewPage() {
       // Read a data page.
       num_buffered_values_ = current_page_header_.data_page_header.num_values;
```

Comment:
Can you elaborate more on the decoder side? Are you planning to process the definition levels in the batch to find the number of non-nulls ?